lib/charms/tls_certificates_interface/v2/tls_certificates.py

```diff
@@ -1592,6 +1660,61 @@ def request_certificate_renewal(
         )
         logger.info("Certificate renewal request completed.")
 
+    def get_assigned_certificates(self) -> List[Dict[str, str]]:
+        """Gets a list of certificates that were assigned to this unit.
+
+        Returns:
+            List of certificates. For example:
+            [
+                {
+                    "ca": "-----BEGIN CERTIFICATE-----...",
+                    "chain": [
+                        "-----BEGIN CERTIFICATE-----..."
+                    ],
+                    "certificate": "-----BEGIN CERTIFICATE-----...",
+                    "certificate_signing_request": "-----BEGIN CERTIFICATE REQUEST-----...",
+                }
+            ]
+        """
+        final_list = []
+        for csr in self.get_certificate_signing_requests(fulfilled_only=True):
+            assert type(csr["certificate_signing_request"]) == str
+            if cert := self._find_certificate_in_relation_data(csr["certificate_signing_request"]):
+                final_list.append(cert)
+        return final_list
+
+    def get_certificate_signing_requests(
+        # RFC: Should I just split this up into 3 different functions?
+        # RFC: Should we have an option where the leader will aggregate all of the CSR's from all of the units of the application?
```

Comment:
I'd provide that option too.

Reply:
I've been investigating how to do this today, and I don't think a unit can read the databag of another unit of the same application unless it's a peer relation, making this impossible. Can you verify?

Reply:
Units can read each other's databags, I just don't know how to iterate over the units of an application, I have been reading a bit too but I didn't find a straightforward way. One could use the unit names like [here](_requirer_csrs), but I don't think that's what we are looking for here.

Reply:
I've talked to the Juju team, and they told me that the preferred way is to look at the units of the peer relation. I will try to see if I can use that.

Reply:
Agreed, if we don't need it, let's not write code for it just in case.